### DISCUSSION TITLE
Tasmota with display - how to show value of HomeAssistant sensor?

### DISCUSSION THREAD

**Discussion by JakubChalupnikPhoenix** [arendst/Tasmota#19596] [Q&A] [answered]
Folks,

I wonder if this is even possible, and I would really appreciate your comments on this.

I have a Tasmota node with a display, connected to Home Assistant. 
My thought was to use the display to show values of some sensors available to Home Assistant. Those sensors are not local to this Tasmota device, but they are MQTT sensors (solar panel power), room temperature (MySensors) etc.

Is that possible? Can Tasmota read sensor values from Home Assistant? If yes, how?
e.g.: my solar panel power is available to HA as "sensor.epever_solar_monitor_panel_p" - how to access it on Tasmota node?
I know this is possible in ESPHome so it's technically doable - or does ESPHome Dashboard handle that?

Thanks a lot!

**Comment by sfromis** [comment, marked as answer]
In general, the structure is not to "read", but that the data source publishes data, which you then can subscribe to. Tasmota already subscribes to commands, making existing commands for the display a possible way of getting data across. Or you can publish commands updating values in Tasmota, and then do what's needed to update the display.

This depends on how you are managing the display. Simplest is the DisplayText etc commands, but with ESP32 you have a lot of more capable options with HASPmota/LVGL and the Berry language improving ways of managing the display. The benefits of this grows with display size....

When it comes to "reading", as initiated from Tasmota.... While this is not a very good strategy, what it would entail is that you'd need to have backend commands to retrieve values, and put them into a response. You cannot "just read" something from another computer on the network, unless you know exactly what type of communication would be used.

As my Home Assistant knowledge is (deliberately) pretty low, I have no answers about how to do things on that platform.

If your sensor devices are publishing via MQTT (like Tasmota is normally doing), you could also do things like subscribe directly to the payloads from the other device(s), or have them publish additional payload specifically for the display device(s). While you may have reasons to go via HA, it may complicate things to need an additional level of processing. Of course, this depends on how your setup is structured.

**Comment by JakubChalupnikPhoenix** [comment]
Thanks, I was kind of worried this will be the case. Some of my sensors don't use MQTT so that will be tricky, but I tried MQTT explorer and I can see some messages related to those sensors, so perhaps HA itself is handling that somehow.
I'll use ESPHome for now and I'll start looking into the whole MQTT stuff when I get some time :)

**Reply by karefor** [reply]
Here it worked via HA indeed. 

Via Einstellungen/Automatisierungen & Szenen Automatisierung erstellen  neue Automatisierung erstellen
choose "when entity changes state or any attributes"
Dann mache / Aktion hinzufügen / (here search for "mqtt") choose "MQTT: Veröffentlichen" 

at Topic I used:    
cmnd/tasmota_200D02/Displaytext    
       
 (for 7-segment Display),((if necessary compiled with gitpod, adding display and mqtt-support lines))
(tasmota_200D02 has to be replaced with the topic name of    y o u r   ESP under Configuration/Configure MQTT the name after %topic% (tasmota_200D02) in the brackets.)

at Nutzlast-Vorlage I used: 
{{ float(states('sensor.tasmota_lk13be_current')) | round(0) }}

Instead of "sensor.tasmota_lk13be_current" you have to use your "sensor.epever_solar_monitor_panel_p" mentioned above.

Good luck!

**Comment by karefor** [comment]
cmnd/tasmota_200D02/Display**number**
and
{{ float(states('sensor.tasmota_lk13be_current')) | round(0) }}**, 0, 0, 4**
makes the ciphers right-aligned
